CMakeLists.txt

```diff
@@ -30,11 +30,42 @@ file(GLOB_RECURSE TVL_GENERATOR_SOURCES CONFIGURE_DEPENDS
 )
 set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${TVL_GENERATOR_SOURCES})
 
+# check for C++20 Concepts features
+if("cxx_std_20" IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+	message("Compiler supports C++20")
+	#set(CMAKE_CXX_STANDARD 20) #this seems like a bad idea
+	set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -std=c++20") #todo: standard should be set in a portable way
+	set(CMAKE_REQUIRED_QUIET True)
+	#unset(SUPPORTS_CONCEPTS CACHE) #we don't need to unset this check
+	INCLUDE(CheckCXXSourceCompiles)
+		CHECK_CXX_SOURCE_COMPILES(
+		[[
+		#include <concepts>
+		struct test_struct {};
+		template<std::copyable T>
+		void test_concepts(T) {};
+		int main(void) {
+			test_concepts(test_struct{});
+			return 0;
+		}
+		]] SUPPORTS_CONCEPTS)
+	if(SUPPORTS_CONCEPTS)
+		message("Compiler does support C++20 and concepts.")
```

Comment:
would recommend `message(STATUS ...)`, these are informational and don't belong on stderr